tls/s2n_handshake_io.c

```diff
@@ -374,8 +374,8 @@ static int handshake_write_io(struct s2n_connection *conn)
     /* Write the handshake data to records in fragment sized chunks */
     struct s2n_blob out;
     while (s2n_stuffer_data_available(&conn->handshake.io) > 0) {
-        int max_payload_size;
-        GUARD((max_payload_size = s2n_record_max_write_payload_size(conn)));
+        int max_payload_size = s2n_record_max_write_payload_size(conn);
+        GUARD(max_payload_size);
```

Comment:
I know we had it as a `GUARD` previously, but I think this makes more sense as `gt_check(max_payload_size, 0)`.